Lib/test/support/__init__.py

```diff
@@ -1877,12 +1883,18 @@ def restore(self):
 
 
 def with_pymalloc():
-    import _testcapi
+    try:
+        import _testcapi
+    except ImportError:
+        raise unittest.SkipTest("requires _testcapi")
     return _testcapi.WITH_PYMALLOC and not Py_GIL_DISABLED
 
 
 def with_mimalloc():
-    import _testcapi
+    try:
+        import _testcapi
+    except ImportError:
```

Comment:
I am a bit worried about what if _testapi raises ImportError with unintended changes and does not notice the important tests are skipped.

Reply:
I didn't take a look the issue deeply but can we skip the test if `--disable-test-modules` is enabled?

Reply:
Or if there is other channel that peope can notice, I think that it's okay.

Reply:
The info is in sysconfig:
```
>>> sysconfig.get_config_var('TEST_MODULES')
'no'
```
On Windows, it`s `None`. I don't know if there's a `--disable-test-modules` equivalent on Windows.
This looks like it could use a helper in `support`.

Reply:
Looking more closely at how these helpers are used, it is not a good idea to simply `raise SkipTest` here. I'll rewrite them to use `sysconfig` instead.

Reply:
Revisiting this PR and the Windows `test_cmd_line` failures, it seems to me `sysconfig` is perhaps not the best route here. I think I prefer my original approach. I'll revert 1169a9536a1e5b8489bbda630dba96180cdc9517 for now.

Reply:
@erlend-aasland: Can you please modify test.pythoninfo to log `sysconfig.get_config_var('TEST_MODULES')` in `collect_sysconfig()`?

Reply:
> I am a bit worried about what if _testapi raises ImportError with unintended changes and does not notice the important tests are skipped.

Overall, it's kind of strange to ask (1) to not build tests and (2) to run tests :-)

Well, I'm in favor of this change, but I'm not surprised that many tests are needed if you explicitly asks to not build tests...

If tomorrow, it becomes a major issue to skip tests in with_mimalloc() because of missing _testcapi, maybe we can consider to expose information about memory allocators in the sys module. But for now, I would prefer to keep these information private.

Reply:
> @erlend-aasland: Can you please modify test.pythoninfo to log `sysconfig.get_config_var('TEST_MODULES')` in `collect_sysconfig()`?

That's a good idea; I'll add it.

Reply:
@vstinner: added in 71968fccb09f52aa2413c40a9154ad88a6c933b5.